packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  try {
+    const response = await fetch(endpoint, {
+      method: "HEAD",
+      cache: "no-cache",
+      mode: "no-cors",
+    });
+    return response.ok;
+  } catch (error) {
+    throw error; // Rethrow for retries
+  }
+}
+
+function useOfflineCheck(options: any = {}) {
+  const {
+    // Optional configuration properties:
+    threshold = 1000, // Minimum interval between offline checks (ms)
+    maxRetries = 3, // Maximum number of retries before assuming offline
+    retryDelay = 1500, // Delay between retries (ms)
+  } = options;
+
+  const [isOffline, setIsOffline] = useState(false);
+  const [retries, setRetries] = useState(0);
+
+  useEffect(() => {
+    const handleOnline = () => {
+      setIsOffline(false);
+      setRetries(0); // Reset retries on reconnection
+    };
+
+    const handleOffline = () => {
+      if (retries < maxRetries) {
+        setTimeout(() => {
+          performNetworkCheck()
+            .then(() => handleOnline()) // online
+            .catch(() => {
+              setRetries(retries + 1); // Retry if error occurs
```

Comment:
I'm quite dubious on the retry logic here, as we don't actually enqueue any retry! The only way retries can be attempted is if the "offline" event fire multiple times. Is that the intention here? But if so, the `retryDelay` variable is a little bit misleading, as it simply adds to the delay between multiple firings of "offline".

Reply:
for the retry logic, we only want to show our app is offline if `retries === maxRetries. so the reason for that.